src/components/Common/UI/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,145 @@
+import type { InputHTMLAttributes, FocusEvent } from 'react'
+import { useRef } from 'react'
+import {
+  DatePicker as AriaDatePicker,
+  Button,
+  Calendar,
+  CalendarCell,
+  CalendarGrid,
+  DateInput,
+  DateSegment,
+  Dialog,
+  Group,
+  Heading,
+  Popover,
+  type DateValue,
+} from 'react-aria-components'
+import { useTranslation } from 'react-i18next'
+import classNames from 'classnames'
+import { CalendarDate } from '@internationalized/date'
+import type { SharedFieldLayoutProps } from '../FieldLayout'
+import { FieldLayout } from '../FieldLayout'
+import { useFieldIds } from '../hooks/useFieldIds'
+import styles from './DatePicker.module.scss'
+import { useTheme } from '@/contexts/ThemeProvider'
+import CaretDown from '@/assets/icons/caret-down.svg?react'
+import CaretRight from '@/assets/icons/caret-right.svg?react'
+import CaretLeft from '@/assets/icons/caret-left.svg?react'
+
+// Utility functions to convert between Date and DateValue
+function dateToDateValue(date: Date | null | undefined): DateValue | null {
+  if (!date) return null
+
+  return new CalendarDate(
+    date.getFullYear(),
+    date.getMonth() + 1, // JavaScript months are 0-indexed
+    date.getDate(),
+  )
+}
+
+function dateValueToDate(dateValue: DateValue | null): Date | null {
+  if (!dateValue) return null
+
+  return new Date(
+    dateValue.year,
+    dateValue.month - 1, // DateValue months are 1-indexed
+    dateValue.day,
+  )
+}
+
+export interface DatePickerProps
+  extends SharedFieldLayoutProps,
+    Pick<InputHTMLAttributes<HTMLInputElement>, 'className' | 'id' | 'name'> {
+  isDisabled?: boolean
+  isInvalid?: boolean
+  onChange?: (value: Date | null) => void
+  onBlur?: (e: FocusEvent) => void
+  label: string
+  value?: Date
+  placeholder?: string
+}
+
+export const DatePicker = ({
+  className,
+  description,
+  errorMessage,
+  id,
+  isDisabled,
+  isInvalid,
+  isRequired,
+  label,
+  onChange,
+  onBlur,
+  value,
+  ...props
+}: DatePickerProps) => {
+  const { t } = useTranslation()
+  const { inputId, errorMessageId, descriptionId, ariaDescribedBy } = useFieldIds({
+    inputId: id,
+    errorMessage,
+    description,
+  })
+  const { container } = useTheme()
+  const dateInputContainerRef = useRef<HTMLDivElement | null>(null)
```

Comment:
Nit: are we using this ref for anything? Seems like it could possibly be removed?

Reply:
Removed